pilot/pkg/ambient/controller/waypoint_controller.go

```diff
@@ -184,6 +189,20 @@ func (rc *WaypointProxyController) Reconcile(name types.NamespacedName) error {
 			ServiceAccount: k,
 			Cluster:        rc.cluster.String(),
 		}
+
+		// Setting non-default mesh config (if set) as the proxy config for the waypoint
+		tmpl, _ := template.New("proxyConfigInject").Funcs(inject.InjectionFuncmap).Parse("{{protoToJSON .}}")
```

Comment:
this seems a bit roundabout. Why not just `             {{ protoToJSON .ProxyConfig }}` in the actual template like we do else-where?